File: src/main/golang/leetcode/editor/cn/[232]Implement Queue using Stacks.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
type MyQueue struct {
	s1 []int
	s2 []int
}

func Constructor() MyQueue {
	return MyQueue{}
}

func (this *MyQueue) Push(x int) {
	this.s2 = append(this.s2, x)
}

func (this *MyQueue) Pop() int {
	num := this.Peek()
	this.s1 = this.s1[:len(this.s1)-1]
	return num
}

func (this *MyQueue) Peek() int {
	if len(this.s1) == 0 {
		for len(this.s2) > 0 {
			curr := this.s2[len(this.s2)-1]
			this.s2 = this.s2[:len(this.s2)-1]
			this.s1 = append(this.s1, curr)
		}
	}
	return this.s1[len(this.s1)-1]
}

func (this *MyQueue) Empty() bool {
	return len(this.s1) == 0 && len(this.s2) == 0
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Peek();
 * param_4 := obj.Empty();
 */
//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[232]Implement Queue using Stacks.py

# leetcode submit region begin(Prohibit modification and deletion)
class MyQueue:

        def __init__(self):
            """
            Initialize your data structure here.
            """
            self.in_stack = []
            self.out_stack = []

        def push(self, x: int) -> None:
            """
            Push element x to the back of queue.
            """
            self.in_stack.append(x)

        def pop(self) -> int:
            """
            Removes the element from in front of queue and returns that element.
            """
            if len(self.out_stack) == 0:
                while len(self.in_stack) > 0:
                    self.out_stack.append(self.in_stack.pop())
            return self.out_stack.pop()

        def peek(self) -> int:
            """
            Get the front element.
            """
            if len(self.out_stack) == 0:
                while len(self.in_stack) > 0:
                    self.out_stack.append(self.in_stack.pop())
            return self.out_stack[-1]

        def empty(self) -> bool:
            """
            Returns whether the queue is empty.
            """
            return len(self.in_stack) == 0 and len(self.out_stack) == 0



# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
# leetcode submit region end(Prohibit modification and deletion)
